### DISCUSSION TITLE
Config issues

### DISCUSSION THREAD

**Discussion by jt1122zz** [andmarti1424/sc-im#893] [General]
I'm using sc-im version 0.8.4 compiled on SparkyLinux (=Debian Testing).

Several issues I've come across:
- binding <C-q> doesn't work: nmap "<C-q>" ":q!<cr>"     
- binding to <C-c> shows "Received SIGINT" so binding isn't recognized 
- :help or :h show a blank screen
- binding to <C-s> doesn't work: nmap "<C-s>" ":w<cr>"  (sc-im hangs until C-c is pressed)
- Is it possible to bind to Alt-<key>? Function key F1-12?

Thanks

**Comment by andmarti1424** [comment]
Its not clear to me what keybinding are you trying to bind. Something like this works without problem:
`nmap "." ":q!<cr>"`
No, you cant bind alt keys on ncurses.
Cannot reproduce your help issue. Seems to be working ok for the dozens of people that use the app. May be there's some issue with your terminal?

**Comment by jt1122zz** [comment]
Thanks for the prompt reply.
Here's the binding I'm try to set:
`nmap "<C-q>" ":q!<cr>"  `
`nmap "<C-s>" ":w<cr>" `

I've tried `:help` both xfce4-terminal and xterm, same result (blank screen).

**Comment by andmarti1424** [comment]
This is not a sc-im issue. I am afraid C-q and C-s and many others keybindings are captured by your shell. You will not be able to bind them in sc-im.
You can try others like C-k C-e

**Reply by jt1122zz** [reply]
Thanks, but I can bind them in other shell apps like nano.
How do I know which C-<key> combination aren't supported in sc-im?
 
Is there a way to bind to F1-F12?

**Comment by andmarti1424** [comment]
I dont know how nano works. it might not even use ncurses.
I believe you cannot map F keys without breaking other stuff. These special keys are captured by your shell as well.
To see which keys or key combinations are supported you might just try and test them.

**Comment by jt1122zz** [comment]
I'm having some issues with some key mappings.
For example, I've defined "cu" for "yank cell" and "cr" for "yank row". They work fine although  both start with "c", i.e. the input waits for a full command to be entered before executing anything.

However defining "hr" for "hide row" doesn't work smoothly and most of the the selection just jumps one cell left due to "h" being already defined and is executed first without waiting for additional inputs. 
I've tried increasing mapping_timeout and command_timeout to 3000 and even 5000 but it doesn't solve the  issue.

Any suggestions? Thanks.
